viamonvif/mdns_server.go

```diff
@@ -82,6 +102,89 @@ func (server *mdnsServer) Remove(hostname string) {
 	delete(server.mappedDevices, hostname)
 }
 
+type cachedEntry struct {
+	DNSName string `json:"dnsname"`
+	IP      string `json:"ip"`
+}
+
+func (server *mdnsServer) readCacheFile() []cachedEntry {
+	// If the `cacheFilepath` was given, read the contents and parse it as json. On any error we log
+	// and return nil, which is interpreted as an empty slice.
+	if server.cacheFilepath == "" {
+		return nil
+	}
```

Comment:
[note to self] cacheFilepath is not given when the discovery service cannot find `VIAM_MODULE_DATA`